src/reducers/importText.ts

```diff
@@ -25,7 +25,7 @@ import sanitize from 'sanitize-html'
 import { getSessionId } from '../util/sessionManager'
 import { ALLOWED_ATTRIBUTES, ALLOWED_TAGS, HOME_PATH } from '../constants'
 import { getAllChildrenAsThoughts } from '../selectors/getChildren'
-import getTextContentFromHTML from '../device/getTextContentFromHTML'
+import * as selection from '../device/selection'
```

Comment:
Reducers need to be pure functions, so we shouldn't have any browser dependencies in this file. Instead, move this up a level to the `importText` action-creator and override the cursor offset with a reducer parameter. It increases the complexity of the reducer signature, but it's worth it to maintain the purity of the function.

Reply:
Yes I got this. So I was thinking overriding the `state.cursorOffset` prior to import text is a bad idea. The reason being it introduces unknown regressions that I faced because of the initial solution I took which included overriding  of `state.cursorOffset`. So what I suggest is instead of using `action-creator` to include the browser selection logic we can add a new payload parameter in `importText` reducer itself. Since we are directly dispatching `importText` reducer from `Editable.tsx` component we don't have to worry about the browser dependencies as well.

Reply:
Sure, that sounds safer!